libraries/lib-components/EffectInterface.h

```diff
@@ -131,10 +138,69 @@ class COMPONENTS_API EffectDefinitionInterface  /* not final */ : public Compone
 
    //! Report names of factory presets
    virtual RegistryPaths GetFactoryPresets() = 0;
+
    //! Change settings to the preset whose name is `GetFactoryPresets()[id]`
    virtual bool LoadFactoryPreset(int id) = 0;
    //! Change settings back to "factory default"
    virtual bool LoadFactoryDefaults() = 0;
+   //! @}
+
+   /*! @name settings
+    Interface for saving and loading externalized settings.
+    All methods are const!
+    */
+   //! @{
+   //! Produce an object holding new, independent settings
+   virtual Settings MakeSettings() const = 0;
+
+   //! Store settings as keys and values
+   /*!
+    @return true on success
+    */
+   virtual bool SaveSettings(
+      const Settings &settings, CommandParameters & parms) const = 0;
+
+   //! Restore settings from keys and values
+   /*!
+    @return true on success
+    */
+   virtual bool LoadSettings(
+      CommandParameters & parms, Settings &settings) const = 0;
+
+   //! @}
+};
+
+//! Extension of EffectDefinitionInterface with old system for settings
+/*!
+ (Default implementations of EffectDefinitionInterface methods for settings call
+ through to the old interface, violating const correctness.  This is meant to be
+ transitional only.)
+ */
+class COMPONENTS_API EffectDefinitionInterfaceEx  /* not final */
+   : public EffectDefinitionInterface
+{
+public:
+   /*! @name old-settings-interface
+    Old interface for saving and loading non-externalized settings
+    */
+   //! @{
+   //! Save current settings into parms
+   virtual bool GetAutomationParameters(CommandParameters & parms) = 0;
+   //! Change settings to those stored in parms
+   virtual bool SetAutomationParameters(CommandParameters & parms) = 0;
```

Comment:
Could we have `const CommandParameters&`  instead? I assume the intent of this method is to only read parms